core/src/main/java/com/exadel/aem/toolkit/bundle/lists/datasource/ChildResourcesDatasource.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.NameConstants;
+import com.adobe.granite.ui.components.Config;
+import com.adobe.granite.ui.components.ExpressionHelper;
+import com.adobe.granite.ui.components.ExpressionResolver;
+import com.adobe.granite.ui.components.PagingIterator;
+import com.adobe.granite.ui.components.ds.AbstractDataSource;
+import com.adobe.granite.ui.components.ds.DataSource;
+import com.adobe.granite.ui.components.ds.EmptyDataSource;
+
+/**
+ * Servlet that implements {@code datasource} pattern for populating a Lists Console
+ * with all child pages under the current root path, which are either lists themselves,
+ * or folders that may contain lists inside
+ */
+@Component(
+    service = Servlet.class,
+    property = {
+        "sling.servlet.resourceTypes=aat-lists/datasources/list",
+        "sling.servlet.methods=" + HttpConstants.METHOD_GET
+    }
+)
+public class ChildResourcesDatasource extends SlingSafeMethodsServlet {
+    private static final String LIST_TEMPLATE_NAME = "/conf/authoring-toolkit/settings/wcm/templates/list";
+    private static final String PN_PATH = "path";
+    private static final String PN_OFFSET = "offset";
+    private static final String PN_LIMIT = "limit";
+    private static final String REP_PREFIX = "rep:";
+    private static final String PN_RESOURCE_TYPE = "itemResourceType";
+    private static final String PATH_TO_JCR_CONTENT = String.format("/%s", JcrConstants.JCR_CONTENT);
+
+    @Reference
+    private transient ExpressionResolver expressionResolver;
+
+    /**
+     * Processes {@code GET} requests to the current endpoint to add to the {@code SlingHttpServletRequest}
+     * a {@code datasource} object filled with all child pages under the current root path, which are either
+     * lists themselves, or folders that may contain lists inside.
+     * The result is then limited by 'offset' and 'limit' parameter values.
+     *
+     * @param request  {@code SlingHttpServletRequest} instance
+     * @param response {@code SlingHttpServletResponse} instance
+     */
+    @Override
+    protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) {
+        ResourceResolver resolver = request.getResourceResolver();
+
+        DataSource dataSource = EmptyDataSource.instance();
+
+        if (getScriptHelper(request) != null) {
+            ExpressionHelper expressionHelper = new ExpressionHelper(expressionResolver, request);
+            Config dsCfg = new Config(request.getResource().getChild(Config.DATASOURCE));
+
+            String parentPath = expressionHelper.getString(dsCfg.get(PN_PATH, String.class));
+            Integer offset = expressionHelper.get(dsCfg.get(PN_OFFSET, String.class), Integer.class);
```

Comment:
Is the use of boxes Integers here justified?

Reply:
Done.